luatest/cbuilder.lua

```diff
@@ -0,0 +1,231 @@
+-- the builder object back).
+--
+-- @usage
+--
+-- local config = Builder:new()
+--     :add_instance('instance-001', {
+--         database = {
+--             mode = 'rw',
+--         },
+--     })
+--     :add_instance('instance-002', {})
+--     :add_instance('instance-003', {})
+--     :config()
+--
+-- By default, all instances are added to replicaset-001 in group-001,
+-- but it's possible to select a different replicaset and/or group:
+--
+-- local config = Builder:new()
+--     :use_group('group-001')
+--     :use_replicaset('replicaset-001')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :use_group('group-002')
+--     :use_replicaset('replicaset-002')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :config()
+--
+-- The default credentials and iproto options are added to
+-- setup replication and to allow a test to connect to the
+-- instances.
+--
+-- There is a few other methods:
+--
+-- * :set_replicaset_option('foo.bar', value)
+-- * :set_instance_option('instance-001', 'foo.bar', value)
+--
+-- @classmod luatest.cbuilder
+
+local checks = require('checks')
+local fun = require('fun')
+
+local Builder = require('luatest.class').new()
+
+-- Do a post-reqiure of the `internal.config.cluster_config`,
+-- since it is available from version 3.0.0+. Otherwise we
+-- will get an error when initializing the module in `luatest.init`.
+local cluster_config = {}
+
+local base_config = {
+    credentials = {
+        users = {
+            replicator = {
+                password = 'secret',
+                roles = {'replication'},
+            },
+            client = {
+                password = 'secret',
+                roles = {'super'},
+            },
+        },
+    },
+    iproto = {
+        listen = {{
+            uri = 'unix/:./{{ instance_name }}.iproto'
+        }},
+        advertise = {
+            peer = {
+                login = 'replicator',
+            }
+        },
+    },
+    replication = {
+        -- The default value is 1 second. It is good for a real
+        -- usage, but often suboptimal for testing purposes.
+        --
+        -- If an instance can't connect to another instance (say,
+        -- because it is not started yet), it retries the attempt
+        -- after so called 'replication interval', which is equal
+        -- to replication timeout.
+        --
+        -- One second waiting means one more second for a test
+        -- case and, if there are many test cases with a
+        -- replicaset construction, it affects the test timing a
+        -- lot.
+        --
+        -- replication.timeout = 0.1 second reduces the timing
+        -- by half for my test.
+        timeout = 0.1,
+    },
+}
+
+function Builder:inherit(object)
+    setmetatable(object, self)
+    self.__index = self
+    return object
+end
+
+function Builder:new(config)
```

Comment:
It is not documented, but should. Also, `config` param should be validated.

Reply:
Added basic documentation for this function.